File: persistence/src/main/java/com/example/solr/data/ProductDocument.java
package com.example.solr.data;

import com.example.solr.data.DetailData;
import org.apache.solr.client.solrj.beans.Field;

import java.util.List;


public class ProductDocument {
    @Field
    public String id;

    @Field
    public List<String> sku;

    @Field(value= "details", child = true)
    public List<DetailData> details;

    public ProductDocument() {

    }

    public ProductDocument(String id, List<String> sku, List<DetailData> detail) {
        this.id = id;
        this.sku = sku;
        this.details = detail;
    }
}

File: persistence/src/main/kotlin/com/example/solr/data/ProductDocument.kt
package com.example.solr.data

import org.apache.solr.client.solrj.beans.Field

class ProductDocument @JvmOverloads constructor(
    @Field
    var id: String? = null,

    @Field
    var sku: String? = null,

    @Field
    var title: String? = null,

    @Field
    var description: String? = null,

    @Field
    var price: Float? = null,

    @Field
    var currency: String? = null,

    @Field("averageprice")
    var averagePrice: Float? = null,

    @Field("lowestprice")
    var lowestPrice: Float? = null,

    @Field("highestprice")
    var highestPrice: Float? = null,

    @Field("medianprice")
    var medianPrice: Float? = null,

    @Field("occurencecount")
    var occurrenceCount: Int? = null,

)
